### PR TITLE
initial implementation

### PR DESCRIPTION
**Pull request by rafikurnia**
Please review, thank you..

### CODE REVIEW

**Comment by andysaputra** [comment]
Should we create a generic module like this one or actually create a module which wraps naming creation for different type of resources?

**Comment by sshvetsov** [comment]
I was thinking the same thing as Andy. Should we make a module specifically for generating unique names for AWS resources (i.e. it knows about the individual resource length constraints and let's user pick the type of resource they need to generate the name for), or let users do the work figuring out what the max length of resource is. Knowing our users, I'm leaning towards former.

**Comment by rafikurnia** [comment]
Got it. I will work on it. Thank you..

File: examples/autoscaling-policy-example/main.tf [hunk 1]
```diff
@@ -0,0 +1,11 @@
+locals {
+  service_name = "txtdata"
+  cluster_role = "app"
+}
+
+module "this" {
+  source = "../../"
+
+  name_prefix   = "${local.service_name}-${local.cluster_role}"
```
Comment (andysaputra) [inline examples/autoscaling-policy-example/main.tf]: missing last hyphen

File: examples/postgres-parameter-group/main.tf [hunk 2]
```diff
@@ -0,0 +1,11 @@
+locals {
+  service_name = "txtdata"
+  db_engine    = "postgres"
+}
+
+module "this" {
+  source = "../../"
+
+  name_prefix   = "${local.service_name}-${local.db_engine}"
```
Comment (andysaputra) [inline examples/postgres-parameter-group/main.tf]: missing last hyphen

Comment (rafikurnia) [reply examples/postgres-parameter-group/main.tf]: The hypen will be added by the module

File: examples/autoscaling-policy-example/main.tf (hunk 1 above)
Comment (rafikurnia) [reply examples/autoscaling-policy-example/main.tf]: The hypen will be added by the module

**Pull request merged**